web/pages/onboarding.tsx

```diff
@@ -95,10 +132,20 @@ const OnboardingPage = (props: Props): ReactElement => {
 
   const onBack = () => {
     if (page.current + 1 > 0) {
+      const previousPage = page.current - 1;
       setPage({
-        current: page.current - 1,
+        current: previousPage,
         previous: page.current,
       });
+      router.push(
```

Comment:
we use this method so often in the code now, maybe we could extract this whole "router shallow push with query" into a function that we can reuse, to clean up/simplify the code a bit?